fast64_internal/oot/c_writer/oot_scene_table_c.py

```diff
@@ -1,75 +1,145 @@
+	for i in range(len(data)):
+		# add a comment to show when it's new scenes
+		if isEndOfExistingScenes:
+			fileData += "// Added scenes\n"
+			isEndOfExistingScenes = False
+		
+		# adds the "// Debug-only scenes" comment after SCENE_GANON_TOU
+		if isEndOfNonDebugScenes:
+			fileData += debugLine
+			isEndOfNonDebugScenes = False
 
-def writeSceneTable(sceneTable, fileData, start, end, exportPath):
-	sceneTableData = sceneTableToString(sceneTable)
-	newFileData = fileData[:start] + sceneTableData + fileData[end:]
+		fileData += f"/* 0x{i:02X} */ DEFINE_SCENE("
+		fileData += ", ".join(str(d) for d in data[i])
 
-	if newFileData != fileData:
-		writeFile(os.path.join(exportPath, 'src/code/z_scene_table.c'), newFileData)
+		if data[i][0] == "testroom_scene":
+			isEndOfExistingScenes = True
+		if data[i][0] == "ganon_tou_scene":
+			isEndOfNonDebugScenes = True
+
+		fileData += ")\n"
+
+	# return the string containing the file data to write
+	return fileData
 
 def modifySceneTable(scene, exportInfo):
+	'''Edit the scene table with the new data'''
 	exportPath = exportInfo.exportPath
-	sceneTable, sceneFileData, start, end = readSceneTable(exportPath)
-	entry = getSceneTableEntryBySceneName(sceneTable, scene.name if scene is not None else exportInfo.name)
+	fileData, header, debugLine = getSceneTable(exportPath)
+	sceneName, sceneTitle, sceneID, sceneUnk10, sceneUnk12, sceneIndex = getSceneParams(scene, exportInfo)
 
-	if scene is not None:
-		if entry is None:
-			sceneTable.append(SceneTableEntry("UNTITLED_SCENE", scene.name + "_scene", None, '0', str(scene.sceneTableEntry.drawConfig), '0'))
-		else:
-			entry.sceneType = entry.sceneType
-			entry.config = str(scene.sceneTableEntry.drawConfig)
+	if scene is None:
+		sceneDrawConfig = None
+	else:
+		sceneDrawConfig = scene.sceneTableEntry.drawConfig
+
+	sceneParams = [sceneName, sceneTitle, sceneID, sceneDrawConfig, sceneUnk10, sceneUnk12]
+
+	# check if it's a custom scene name
+	if sceneIndex is None: 
+		isCustom = True
 	else:
-		if entry is not None:
-			sceneTable.remove(entry)
-	writeSceneTable(sceneTable, sceneFileData, start, end, exportPath)
\ No newline at end of file
+		isCustom = False
+
+	# if so, check if the custom scene already exists in the data
+	# if it already exists set isCustom to False to consider it like a normal scene
+	if isCustom:
+		for i in range(len(fileData)):
+			if fileData[i][0] == scene.name.lower() + "_scene":
+				sceneIndex = i
+				isCustom = False
+				break
+
+	# edit the current data or append new one if we are in a ``Custom`` context
+	for i in range(6):
+		if isCustom == False and sceneParams[i] is not None and fileData[sceneIndex][i] != sceneParams[i]:
+			fileData[sceneIndex][i] = sceneParams[i]
+		elif isCustom:
+			fileData.append(sceneParams)
+			sceneIndex = len(fileData) - 1
+			break
+
+	# remove the scene data if scene is None (`Remove Scene` button)
+	if scene is None:
+		if sceneIndex < len(fileData):
```

Comment:
mb on the suggested change here also need
```suggestion
		if sceneIndex is not None and sceneIndex < len(fileData):
```